salt/core/ntp.sls

```diff
@@ -1,4 +1,24 @@
-# Disable pre-installed systemd-timesyncd as we prefer NTP.
+{% if grains['osrelease']|float >= 20.04 %}
+# systemd-timesyncd should be installed by default unless the NTP set up code below has ran against server.
+systemd-timesyncd:
+  pkg.installed:
+    - name: systemd-timesyncd
+  service.running:
+    - enable: True
+
+/etc/systemd/timesyncd.conf:
+  file.replace:
+    - pattern: "^#?NTP=.*"
+    - repl: 'NTP=0.uk.pool.ntp.org 1.uk.pool.ntp.org 2.uk.pool.ntp.org 3.uk.pool.ntp.org/'
+    - watch_in:
+      - service: systemd-timesyncd
+
+# Catch instances where ntp has been installed previously.
+ntp:
+  service.dead:
+    - enable: False
+
+{% else %}
```

Comment:
Why is the service disabled on earlier Ubuntu versions?

Reply:
systemd-timesyncd has been pre-installed by ubuntu for a few releases now.
For a long time we preferred using NTP but we have internally agreed that full NTP is overkill for most servers. 
systemd-timesyncd is a simpler implementation of the NTP protocol syncing time to one server rather than against multiple NTP endpoints.

The one thing we need to be careful around is ending up with two different systems being installed both changing the system time.
This PR swaps it around the install to prefer systemd-timesyncd and disable NTP - in the current state of this PR this should only apply to amy01 where NTP has been installed by us already.


Considering your other comments, it'd be nice to use the same configuration across versions, so we could remove the if statement completely here.

Reply:
I'm happy with removing the if statement :)